OrbitGl/ThreadTrack.cpp

```diff
@@ -27,23 +36,223 @@ void ThreadTrack::Draw(GlCanvas* a_Canvas, bool a_Picking) {
   // TODO: Make a proper "SchedTrack" instead of hack.
   if (m_ID == 0) return;
 
-  TimeGraphLayout& layout = m_TimeGraph->GetLayout();
-  float threadOffset = layout.GetThreadBlockStart(m_ID);
   float trackHeight = GetHeight();
   float trackWidth = a_Canvas->GetWorldWidth();
 
-  SetPos(a_Canvas->GetWorldTopLeftX(), threadOffset);
+  SetPos(a_Canvas->GetWorldTopLeftX(), m_Pos[1]);
   SetSize(trackWidth, trackHeight);
 
   Track::Draw(a_Canvas, a_Picking);
 
   // Event track
   m_EventTrack->SetPos(m_Pos[0], m_Pos[1]);
   m_EventTrack->SetSize(a_Canvas->GetWorldWidth(),
-                        m_TimeGraph->GetLayout().GetEventTrackHeight());
+                        time_graph_->GetLayout().GetEventTrackHeight());
   m_EventTrack->Draw(a_Canvas, a_Picking);
 }
 
+//-----------------------------------------------------------------------------
+std::string GetExtraInfo(const Timer& a_Timer) {
+  std::string info;
+  static bool show_return_value = absl::GetFlag(FLAGS_show_return_values);
+  if (!Capture::IsCapturing() && a_Timer.GetType() == Timer::UNREAL_OBJECT) {
+    info =
+        "[" + ws2s(GOrbitUnreal.GetObjectNames()[a_Timer.m_UserData[0]]) + "]";
+  } else if (show_return_value && (a_Timer.GetType() == Timer::NONE)) {
+    info = absl::StrFormat("[%lu]", a_Timer.m_UserData[0]);
+  }
+  return info;
+}
+
+//-----------------------------------------------------------------------------
+void ThreadTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick) {
+
+  m_EventTrack->UpdatePrimitives(min_tick, max_tick);
+  Batcher* batcher = &time_graph_->GetBatcher();
+  TextRenderer* text_renderer = time_graph_->GetTextRenderer();
+  GlCanvas* canvas = time_graph_->GetCanvas();
+
+  const TimeGraphLayout& m_Layout = time_graph_->GetLayout();
+  const TextBox& m_SceneBox = canvas->GetSceneBox();
+  float minX = m_SceneBox.GetPosX();
+
+  double m_TimeWindowUs = time_graph_->GetTimeWindowUs();
+  float m_WorldStartX = canvas->GetWorldTopLeftX();
+  float m_WorldWidth = canvas->GetWorldWidth();
+  double invTimeWindow = 1.0 / m_TimeWindowUs;
+
+  std::vector<std::shared_ptr<TimerChain>> depthChain = GetTimers();
+  for (auto& textBoxes : depthChain) {
+    if (textBoxes == nullptr) break;
+
+    for (TextBox& textBox : *textBoxes) {
+      const Timer& timer = textBox.GetTimer();
+
+      if (!(min_tick > timer.m_End || max_tick < timer.m_Start)) {
+        double start = time_graph_->GetUsFromTick(timer.m_Start);
+        double end = time_graph_->GetUsFromTick(timer.m_End);
+        double elapsed = end - start;
+
+        double NormalizedStart = start * invTimeWindow;
```

Comment:
Variable naming is inconsistent. I guess this is copy/pasted from somewhere else and then modified. Up to you how far you want to fix naming to be consistent.

Reply:
This was indeed copied.  There are quite a few to change, I wanted to refactor that code anyway, will do it in next PR.